packages/server/tests/acceptance/hbarLimiter.spec.ts

```diff
@@ -220,36 +220,49 @@ describe('@hbarlimiter HBAR Limiter Acceptance Tests', function () {
         global.accounts.push(...accounts);
       });
 
-      beforeEach(async function () {
-        const basicPlans = await hbarSpendingPlanRepository.findAllActiveBySubscriptionTier(
-          [SubscriptionTier.BASIC],
-          requestDetails,
-        );
-        for (const plan of basicPlans) {
-          await hbarSpendingPlanRepository.delete(plan.id, requestDetails);
-          await ethAddressSpendingPlanRepository.deleteAllByPlanId(plan.id, 'before', requestDetails);
-          await ipSpendingPlanRepository.deleteAllByPlanId(plan.id, 'before', requestDetails);
-        }
+      afterEach(async () => {
+        hbarSpendingPlanRepository.resetAmountSpentOfAllPlans(requestDetails);
+
+        // Note: Since the total HBAR budget is shared across the entire Relay instance by multiple test cases,
```

Comment:
Q: Are we running multiple instances in acceptance tests or are you referring to parallel runs?
If parallel aren't those instances separate from each other?

Reply:
Ah we only run one instance of the Relay in the suite. The issue is that the tests in the suite share and depend on the total HBAR budget controlled by the `HbarLimitService` class. All expense additions occur asynchronously by emitting an event [like this](https://github.com/hashgraph/hedera-json-rpc-relay/blob/main/packages/relay/src/lib/clients/sdkClient.ts#L747-L755) and catching the event in the `MetricService` class [like this](https://github.com/hashgraph/hedera-json-rpc-relay/blob/main/packages/relay/src/lib/services/metricService/metricService.ts), which happens asynchronously in the background to avoid unnecessary wait time.

Reply:
Hmm, okay.
This is the impact of the async emission of events I guess